File: ecmascript/unique/2-unique.js
function unique(list) {
    return Array.from(new Set(list));
}

function main() {
    let list = [1, 2, 3, 4, 5, 6];
    console.log({list, unique: unique(list)});
    list = [1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6];
    console.log({list, unique: unique(list)});
    list = [1, 6, 5, 4, 3, 2, 1, 2, 3, 4, 5, 6];
    console.log({list, unique: unique(list)});
}
main();

File: rust/unique/2-unique.rs
fn unique(list: &mut Vec<i32>) -> Vec<i32> {
    list.sort();
    list.dedup();
    list.to_vec()
}


fn main() {
    let mut list = vec![1, 2, 3, 4, 5, 6];
    println!("original: {:?}, unique: {:?}", list, unique(&mut list.clone()));
    list = vec![1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 5, 6];
    println!("original: {:?}, unique: {:?}", list, unique(&mut list.clone()));
    list = vec![1, 6, 5, 4, 3, 2, 1, 2, 3, 4, 5, 6];
    println!("original: {:?}, unique: {:?}", list, unique(&mut list.clone()));
}
